source/common/network/connection_impl.cc

```diff
@@ -140,7 +140,7 @@ void ConnectionImpl::close(ConnectionCloseType type) {
   const bool delayed_close_timeout_set = delayed_close_timeout_.count() > 0;
   if (data_to_write == 0 || type == ConnectionCloseType::NoFlush ||
       !transport_socket_->canFlushClose()) {
-    if (data_to_write > 0) {
+    if (data_to_write > 0 && type != ConnectionCloseType::NoFlush) {
       // We aren't going to wait to flush, but try to write as much as we can if there is pending
       // data.
       transport_socket_->doWrite(*write_buffer_, true);
```

Comment:
thanks for a quick update! i was trying to see if we have any current unit breakage. 
i'm not super sure what you mean about the else block.
we want to prevent `doWrite`  for NoFlush even if we have data, right?